e2e/node/e2e-test.spec.ts

```diff
@@ -64,6 +72,131 @@ function getCredentials() {
   };
 }
 
+describe("handleIncomingRedirect", () => {
```

Comment:
> using jest as a node runner but a plain script (potentially using express) that implements the server-side, and have playwright drive the test against that.

Good idea - I've implemented this in https://github.com/inrupt/solid-client-authn-js/pull/3044/commits/41a7c75feeca46fc50309fe5e6ae0ae636b55d4e .

However, I think we should have this in addition to the original node test as they test two distinct use cases (the original testing scripts with browser interaction components to login; the second which I just added testing apps with server-side rendering).

Would moving these tests to a new file resolve your concerns?

Reply:
Having these tests separate would be an improvement, but I'm challenging whether we need the former at all. Is "scripts with browser interaction components to login" a use case that is important enough that we add the complexity of this test to cover it, while it will rely on the same code paths as a regular Node web server? Because essentially, what happens is:
- The Node code calls `login`, and redirects the user away to the OP via their browser.
- The OP eventually redirects the user back to the node code with the redirect URL containing the auth code

I don't think embedding playwright in the Node tests is worth it. Having an example of script that does that type of redirect, why not, but having a full-fledged browser testing framework on top of jest just for this niche use case is overkill in my opinion.

Reply:
> "scripts with browser interaction components to login" a use case that is important enough that we add the complexity of this test to cover it

I've encountered this several times in the wild; in particular:
 - A similar flow is used to log into desktop applications like SolidFS; and can be used for logging into slack-style apps in the future.
 - A similar flow is being used by SolidLab to log into browser extensions
 - I've seen multiple people doing experiments with Solid are logging in with scripts this way; especially if they want to test against other Solid Servers that don't support client-creds in addition to testing against the ESS.

> while it will rely on the same code paths as a regular Node web server

I agree with this; the main differences between the two is that the browser is closed and re-opened between login and logout; and the app is redirects to the login/logout pages rather than opening the browser directly on those pages.

> overkill in my opinion.

I was viewing this in terms of whether the ongoing maintenance cost of this test would be worth it; I was imagining this would be fairly low since it is re-using existing components from the browser tests.

For now I will move this to a separate file; and we can remove these tests once you're back from break if they are still overkill / not worth the maintenance cost.